cognite/client/_api/datapoint_tasks.py

```diff
@@ -1036,40 +1095,49 @@ def _find_number_of_subtasks_uniform_split(self, tot_ms: int, n_workers_per_quer
         super().__init__(query=query, use_numpy=use_numpy, **kwargs)
 
     @cached_property
     def offset_next(self) -> int:
         return granularity_to_ms(cast(str, self.query.granularity))
 
-    def _set_aggregate_vars(self, aggs_camel_case: list[str], use_numpy: bool) -> None:
+    def _set_aggregate_vars(self, aggs_camel_case: list[str], use_numpy: bool, include_status: bool) -> None:
         # Developer note here: If you ask for datapoints to be returned in JSON, you get `count` as an integer.
-        # Nice. However, when using protobuf, you get `double` xD ...so when this code was pivoted from
-        # JSON -> protobuf, the special handling of `count` was kept in the hopes that one day protobuf
-        # would yield the correct type...!
+        # Nice. However, when using protobuf, you get `double` xD
         self.all_aggregates = aggs_camel_case
-        self.int_aggs = _INT_AGGREGATES.intersection(aggs_camel_case)
-        self.float_aggs = set(aggs_camel_case).difference(self.int_aggs)
-
-        self.agg_unpack_fn = DpsUnpackFns.custom_from_aggregates(self.all_aggregates)
-        self.first_agg, *others = self.all_aggregates
-        self.single_agg = not others
-
-        if use_numpy:
-            if self.single_agg:
-                self.dtype_aggs: np.dtype[Any] = np.dtype(np.float64)
-            else:  # (.., 1) is deprecated for some reason
-                # We are storing all (2 -> 16) aggregates in one block of memory:
-                self.dtype_aggs = np.dtype((np.float64, len(self.all_aggregates)))
+        self.object_aggs = list(_OBJECT_AGGREGATES.intersection(aggs_camel_case))
+        if self.object_aggs:
+            self.object_data: dict[Literal["minDatapoint", "maxDatapoint"], _DataContainer] = {
+                agg: defaultdict(list) for agg in self.object_aggs
+            }
+            self.object_agg_unpack_fns = [
+                DpsUnpackFns.extract_fn_min_or_max_dp(agg, include_status) for agg in self.object_aggs
+            ]
+        self.numeric_aggs = [agg for agg in aggs_camel_case if agg not in self.object_aggs]
+        self.n_numeric_aggs = len(self.numeric_aggs)
+        if self.n_numeric_aggs:
+            self.numeric_agg_unpack_fn = DpsUnpackFns.custom_from_aggregates(self.numeric_aggs)
+            if use_numpy:
+                if self.n_numeric_aggs == 1:
+                    self.dtype_aggs: np.dtype[Any] = np.dtype(np.float64)
+                else:  # (.., 1) is deprecated for some reason
+                    # We are storing all (2 -> 16) aggregates in one block of memory:
+                    self.dtype_aggs = np.dtype((np.float64, self.n_numeric_aggs))
+
+        # We must unpack all data as double (see dev. note above), but we need to know which should be cast to int:
+        self.int_aggs = _INT_AGGREGATES.intersection(self.numeric_aggs)
+        self.float_aggs = [agg for agg in self.numeric_aggs if agg not in self.object_aggs]
```

Comment:
```suggestion
        self.float_aggs = [agg for agg in self.numeric_aggs if agg not in self.int_aggs]
```

Reply:
No, that’s not correct. The comment two lines above explains why (they must still be unpacked as floats as that is how protobuf stores them)

Reply:
Fixed in 8a13fc1e8401df7e74df61fafe4134103e27ee83